src/cascadia/TerminalSettingsEditor/Keybindings.cpp

```diff
@@ -0,0 +1,238 @@
+    {
+        Controls::TextBox textBox = sender.as<Controls::TextBox>();
+
+        if (e.Key() == VirtualKey::Back)
+        {
+            m_keysInBind.clear();
+        }
+        else
+        {
+            m_keysInBind.insert(e.Key());
+            if (textBox != nullptr)
+            {
+                textBox.Text(GetKeyListString());
+            }
+        }
+
+        e.Handled(true);
+    }
+
+    void Keybindings::KeyBindTextBox_TextChanging(winrt::Windows::UI::Xaml::Controls::TextBox const& sender, winrt::Windows::UI::Xaml::Controls::TextBoxTextChangingEventArgs const& /*args*/)
+    {
+        hstring currText = sender.Text();
+        hstring newText = hstring(currText.data(), currText.size());
+
+        sender.Text(newText);
+    }
+
+    void Keybindings::ShowOptionsButtonIfRequired(hstring tag)
+    {
+        std::set<hstring> settingsWithOptions;
+        settingsWithOptions.insert(c_openSettingsTag);
+        settingsWithOptions.insert(c_newTabTag);
+        settingsWithOptions.insert(c_switchToTabTag);
+        settingsWithOptions.insert(c_renameTabTag);
+        settingsWithOptions.insert(c_setTabColorTag);
+        settingsWithOptions.insert(c_moveFocusTag);
+        settingsWithOptions.insert(c_resizePaneTag);
+        settingsWithOptions.insert(c_splitPaneTag);
+        settingsWithOptions.insert(c_copyTag);
+
+        Windows::UI::Xaml::Visibility expectedVisibility = Visibility::Collapsed;
+        if (settingsWithOptions.find(tag) != settingsWithOptions.end())
+        {
+            expectedVisibility = Visibility::Visible;
+        }
+        m_optionalSettingsPanel.Visibility(expectedVisibility);
+        m_addNewButton.Visibility(expectedVisibility);
+        if (m_lastOpenedArgsPanel != nullptr)
+        {
+            m_lastOpenedArgsPanel.Visibility(Visibility::Collapsed);
+        }
+    }
+
+    void Keybindings::CommandComboBox_SelectionChanged(winrt::Windows::Foundation::IInspectable const& sender, winrt::Windows::UI::Xaml::Controls::SelectionChangedEventArgs const& /*e*/)
+    {
+        hstring selectedItemTag = GetSelectedItemTag(sender);
+        ShowOptionsButtonIfRequired(selectedItemTag);
+    }
+
+    void Keybindings::SaveButton_Click(winrt::Windows::Foundation::IInspectable const& /*sender*/, winrt::Windows::UI::Xaml::RoutedEventArgs const& /*e*/)
+    {
+        // Collect the information here
+        winrt::Windows::UI::Popups::MessageDialog msg{ CollectInputData() };
+        msg.ShowAsync();
+    }
+
+    hstring Keybindings::TraversePanel(const Controls::Panel& panel)
+    {
+        hstring fullInfo;
+
+        for (const auto panelChild : panel.Children())
+        {
+            if (Controls::ComboBox childComboBox = panelChild.try_as<Controls::ComboBox>())
+            {
+                fullInfo = fullInfo + childComboBox.Name() + L":" + GetSelectedItemTag(childComboBox);
```

Comment:
We could probably use the `fmt` lib to replace all the string building - here and everywhere else?

Reply:
For now I'll put a TODO and a reminder in the epic to do this as we go, I expect a lot of the strings in the proj to get changed.